src/lua/utils.c

```diff
@@ -422,7 +422,8 @@ luaT_newmodule(struct lua_State *L, const char *modname,
 	lua_newtable(L);
 
 	/* Fill the module table with functions. */
-	luaL_setfuncs(L, funcs, 0);
+	if (funcs != NULL)
+		luaL_setfuncs(L, funcs, 0);
```

Comment:
AFAIU you should use `luaT_setmodule` instead of patching `luaT_newmodule`.

Reply:
We use `luaT_newmodule()` for a new module created from C. I think it is OK to use it here as well -- the case is pretty same.

However, it is different in that we don't want to register any function. This is the first such case, however I don't see anything wrong here. Just a module without functions.